ToDoApp/ToDoApp/ToDoService.swift

```diff
@@ -0,0 +1,61 @@
+//
+//  ToDoDetail.swift
+//  ToDoApp
+//
+//  Created by Trong Hieu Vuong on 2022-11-29.
+//
+
+import Foundation
+
+enum TaskError: Error {
+    case unavailableTask
+}
+
+struct Task {
+    var id: UUID = UUID()
+    var title: String = "New Task"
+    var note: String = "New Note"
+    var deadline: Date?
+    var status: Bool = false
+}
+
+protocol ToDoService {
+    func createTask(title: String, note: String, deadline: Date?) async -> Result<Task, TaskError>
+    func updateTask(todo: Task) async -> Result<Task, TaskError>
+    func taskForToday() -> [Task]
+    func checkIsToday(date: Date?) -> Bool
```

Comment:
this method is not needed to be in the ToDoService protocol, since clients of the ToDoService don't need to invoke this method, and it should be private in the FeaturesToDo class